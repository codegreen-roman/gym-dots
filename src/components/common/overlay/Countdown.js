import React from 'react'
import { keyframes } from 'emotion'
import styled from 'emotion/react'

const CountdownAnimation = keyframes`
    0% {text-indent:0}
    100% {text-indent:-3ch;}
`

const CountdownStyled = styled.div`
    &:before {
        content: '321';
        font-size: 40px;
        display: inline-block;
        width: 1ch;
        overflow: hidden;
        animation: ${CountdownAnimation} 3s steps(3) forwards;
    }
`

export const Countdown = () => {
    return <CountdownStyled />
}
