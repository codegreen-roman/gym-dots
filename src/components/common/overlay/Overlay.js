import React from 'react'
import { OverlayStyled } from './Overlay.emo'
import { bool, element } from 'prop-types'

export const Overlay = ({ visible, children }) => {
    if (!visible) return null
    return (
        <OverlayStyled visible>
            {children}
        </OverlayStyled>
    )
}

Overlay.propTypes = {
    visible: bool.isRequired,
    children: element.isRequired
}
