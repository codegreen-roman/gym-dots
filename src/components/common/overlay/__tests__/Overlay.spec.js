import React from 'react'
import { shallow } from 'enzyme'
import { Overlay } from '../Overlay'
import { Countdown } from '../Countdown'

const setup = (visible = false) => {
    const props = {
        visible,
        children: <Countdown />
    }

    return {
        wrapper: shallow(<Overlay {...props} />),
        props
    }
}

describe('Overlay Component', () => {
    it('should be hidden by default', () => {
        const { wrapper } = setup()
        expect(wrapper).toMatchSnapshot()
    })

    it('should be visible on the screen', () => {
        const { wrapper } = setup(true)
        expect(wrapper).toMatchSnapshot()
    })

    it('should be hidden on the screen', () => {
        const { wrapper } = setup(false)
        expect(wrapper).toMatchSnapshot()
    })
})
