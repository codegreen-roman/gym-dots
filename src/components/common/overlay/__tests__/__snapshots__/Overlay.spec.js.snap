// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Component should be hidden by default 1`] = `null`;

exports[`Overlay Component should be hidden on the screen 1`] = `null`;

exports[`Overlay Component should be visible on the screen 1`] = `
<styled(Component)
  visible={true}
>
  <Countdown />
</styled(Component)>
`;
